plutus-core/plutus-ir/test/transform/inline/letOverAppType.golden

```diff
@@ -3,5 +3,8 @@
   (termbind
     (strict) (vardecl idFun (all a (type) (fun a a))) (abs a (type) (lam x a x))
   )
-  (con integer 3)
+  [
```

Comment:
regression?

Reply:
Nothing got through to `applyAndBetaReduce` in this case - that means that `k` and `funApp` got unconditionally inlined, and they were never added to the variable map, so `inlineApp` went to the `otherwise` branch. I have removed call site inlining in https://github.com/input-output-hk/plutus/commit/d4f68656860b834c813af5080f5d6f7a9d4ab45a and verified that `k` did indeed get inlined by unconditional inlining. I'll update the comment in the test.

Reply:
This test case is supposed to test over-application (as its name suggests). If everything is unconditionally inlined, then we aren't testing that, and so the test case should either be removed or updated. I wonder why it wasn't unconditionally inlined before, but it is now - there should be no change to the unconditionally inlining condition?